BUILD.md

```diff
@@ -0,0 +1,86 @@
+
+It may be easier to set up one over the other, depending on which flavour of Linux is in use.
+
+> ### Podman  
+>  
+> According to your distribution's particular package manager, install `podman`.  
+> `# pacman -S podman`  
+> `# apt-get install podman`  
+> `...`  
+
+Information on setting up `podman` to allow rootless container deployment can be found [here](https://github.com/containers/podman/blob/main/docs/tutorials/rootless_tutorial.md).
+
+> ### Docker  
+>  
+> According to your distribution's particular package manager, install `docker`...  
+>  
+> `# pacman -S docker`  
+> `# apt-get install docker`  
+> `...`  
+>  
+> ...and `-a`ppend any desired usernames to the `docker` user`-G`roup, to allow these users to work with images and containers *sans root.*  
+>  
+> `# usermod -a -G docker <jrtoddy>`  
+
+## Obtaining the Sources
+
+The Git `clone` command will create a copy of the current ZFSBootMenu source hierarchy in a chosen target `<directory>`; by default, this will be `./zfsbootmenu` if no other argument is given.
+
+> `$ git clone https://github.com/zbm-dev/zfsbootmenu.git <directory>`  
+
+Likewise, a tarball can be obtained and extracted using normal command line tools.
+
+> `$ wget https://github.com/zbm-dev/zfsbootmenu/archive/master.tar.gz`  
+> `$ tar -xvf master.tar.gz`  
+
+## The Process
+
+The ZFSBootMenu build process is initiated by the `zbm-builder.sh` script in the root of the repository.
+
+It assumes that the local system is the target, and, as such, it attempts to fetch a copy of the local [`/etc/zfs/zpool.cache`](https://openzfs.github.io/openzfs-docs/Project%20and%20Community/FAQ.html#the-etc-zfs-zpool-cache-file) and [`/etc/hostid`](https://openzfs.github.io/openzfs-docs/man/8/zgenhostid.8.html) to be integrated into the `initramfs`.
+
+This behaviour can be disabled by invoking `zbm-builder.sh` with the `-C` and `-H` arguments, to disable use of the local `C`ache and `H`ostid respectively.
+
+It will then launch a temporary container via [`docker`](http://www.docker.com) or [`podman`](https://www.podman.io) using the image `ghcr.io/zbm-dev/zbm-builder:latest`. 
+
+Use the `-B` argument to specify another `B`uild environment. Information for building your own custom build environment can be found [above](https://github.com/zbm-dev/zfsbootmenu/blob/master/releng/docker/README.md).
+
+On creation, the entry-point script `zbm-build.sh` will be executed inside the instance. It will pull a copy of the current `master` source tree from ZFSBootMenu's Github repository, and build.
+
+If building of an alternate branch is required, it can be specified with the `-t`ag command line option; likewise, using the `-L` option will allow the use of `L`ocal sources in the directory in which `zbm-builder.sh` script is located instead.
```

Comment:
It might be nice to give an example of a specific tag - such as `v1.12.0`. A link to https://github.com/zbm-dev/zfsbootmenu/tags could also be embedded.

It'd be nice to see aan explanation on how you can define a local `config.yaml` that takes precedence over the default that is in the builder container. A link to documentation on that would be good as well.

Reply:
No problem, will do. It was something that occurred to me also.